atxm/strategies.py

```diff
@@ -136,57 +138,147 @@ def execute(self, pending: PendingTx) -> TxParams:
                 fault=Fault.TIMEOUT,
                 message="Transaction has timed out",
             )
-        return pending.params
+        return None
 
 
-class FixedRateSpeedUp(AsyncTxStrategy):
-    """Speedup strategy for pending transactions."""
+class ExponentialSpeedupStrategy(AsyncTxStrategy):
+    """
+    Speedup strategy for pending transactions that increases fees by a
+    percentage over the prior value every time it is used.
+    """
 
-    SPEEDUP_FACTOR = 1.125  # 12.5% increase
-    MAX_TIP = Gwei(1)  # gwei maxPriorityFeePerGas per transaction
+    _SPEEDUP_INCREASE_PERCENTAGE = 0.125  # 12.5%
 
-    _NAME = f"speedup-{SPEEDUP_FACTOR}%"
+    _MIN_SPEEDUP_INCREASE = 0.10  # mandated by eth standard
 
-    def _calculate_speedup_fee(self, pending: TxParams) -> Tuple[Wei, Wei]:
-        base_fee = self.w3.eth.get_block("latest")["baseFeePerGas"]
+    _MAX_TIP_FACTOR = 3  # max 3x over suggested tip
+
+    _NAME = "speedup"
+
+    _GAS_PRICE_FIELD = "gasPrice"
+    _MAX_FEE_PER_GAS_FIELD = "maxFeePerGas"
+    _MAX_PRIORITY_FEE_PER_GAS_FIELD = "maxPriorityFeePerGas"
+
+    def __init__(
+        self,
+        w3: Web3,
+        speedup_increase_percentage: float = _SPEEDUP_INCREASE_PERCENTAGE,
+        max_tip_factor: int = _MAX_TIP_FACTOR,
+    ):
+        super().__init__(w3)
+
+        if (
+            speedup_increase_percentage < self._MIN_SPEEDUP_INCREASE
+            or speedup_increase_percentage > 1
+        ):
+            raise ValueError(
+                f"Invalid speedup increase percentage {speedup_increase_percentage}; "
+                f"must be in range [0.10, 1]"
+            )
+        if max_tip_factor <= 1:
+            raise ValueError(f"Invalid max tip factor {max_tip_factor}; must be > 1")
+
+        self.speedup_factor = 1 + speedup_increase_percentage
+        self.max_tip_factor = max_tip_factor
+
+    def _calculate_eip1559_speedup_fee(self, params: TxParams) -> Tuple[int, int, int]:
```

Comment:
@vzotova , @cygnusv pay particular attention to this logic, and the logic in `_calculate_legacy_speedup_fee()`.